recipe/test_nvcc.sh

```diff
@@ -64,8 +64,49 @@ if [[ "${CFLAGS}" == "${CFLAGS_CONDA_NVCC_TEST}" ]]; then
 else
     echo "CFLAGS is incorrectly set to '${CFLAGS}', should be set to '${CFLAGS_CONDA_NVCC_TEST}'" && exit 1
 fi
+
 # Reactivate
 source ${PREFIX}/etc/conda/activate.d/${PKG_NAME}_activate.sh
 
 # Try building something
-nvcc test.cu
\ No newline at end of file
+nvcc test.cu
+
+# Try different CMake setups
+cd cmake-tests/
+
+set +ex
+exitcode=0
+count=0
+summary=""
+for flags in \
+    "-DWITH_ENABLE_LANGUAGE=OFF -DCUDA_FINDER=CUDA       " \
+    "-DWITH_ENABLE_LANGUAGE=OFF -DCUDA_FINDER=CUDAToolkit" \
+    "-DWITH_ENABLE_LANGUAGE=ON  -DCUDA_FINDER=CUDA       " \
+    "-DWITH_ENABLE_LANGUAGE=ON  -DCUDA_FINDER=CUDAToolkit" \
+    "-DWITH_ENABLE_LANGUAGE=ON  -DCUDA_FINDER=OFF        " \
+    ; do
+        ((count+=1))
+        echo -e "\n\n--------"
+        echo -e "Test #$count:" $flags
+        echo -e "--------\n\n"
+        rm -rf build || true
+        mkdir -p build
+        cd build
+        export CUDACXX=${CUDA_PATH}/bin/nvcc
+        export CUDAHOSTCXX=${CC}
```

Comment:
We absolutely shouldn't need to set these, otherwise every downstream project would need to potentially set these as well.

Reply:
This is what I needed to pass to make CMake build the project. We need to decide whether we move this to the activation script or not.

Reply:
I've moved it for now, I don't think it'll hurt having it there, but happy to change it if needed!